src/dependency_groups/__main__.py

```diff
@@ -9,43 +9,62 @@
 except ImportError:
     try:
         import tomli as tomllib  # type: ignore[no-redef]
-    except ImportError:
-        tomllib = None  # type: ignore[assignment]
+    except ModuleNotFoundError:
+        print(
+            "Usage error: dependency-groups CLI requires tomli or Python 3.11+",
+            file=sys.stderr,
+        )
+        raise SystemExit(2) from None
```

Comment:
Just curious, does explicitly raising a SystemExit behave significantly differently from `sys.exit`? (Will it capture the outer error in `__cause__` by default?)

If I need to adjust my habits around this, it's a good opportunity to learn.

Reply:
It doesn't require an import, is microscopically faster, and is a bit more explicitly obvious as a control flow statement than a NoReturn function. If someone does explicitly catch it, like in tests, I think can then inspect the `__cause__`, as well, but it doesn't show the cause with the default exception hook.

See https://www.youtube.com/watch?v=ZbeSPc5wL0g.


Oh, interesting, you do have to explicitly use a `from e` here. I wonder if that's special cased?

```pycon
>>> try:
...     try:
...         1 / 0
...     except ZeroDivisionError:
...         raise SystemExit(1)
... except SystemExit as e:
...     print(e.__cause__)
...
None
>>> try:
...     try:
...         1 / 0
...     except ZeroDivisionError as e:
...         raise SystemExit(1) from e
... except SystemExit as e:
...     print(e.__cause__)
...
division by zero
```